clients/src/main/java/org/apache/kafka/clients/admin/MemberAssignment.java

```diff
@@ -34,10 +33,10 @@
      * Creates an instance with the specified parameters.
      *
      * @param topicPartitions List of topic partitions
+     * @throws NullPointerException if topicPartitions is null
      */
     public MemberAssignment(Set<TopicPartition> topicPartitions) {
-        this.topicPartitions = topicPartitions == null ? Collections.<TopicPartition>emptySet() :
-            Collections.unmodifiableSet(new HashSet<>(topicPartitions));
+        this.topicPartitions = Collections.unmodifiableSet(new HashSet<>(topicPartitions));
```

Comment:
It's customary to use `Objects.requireNonNull(topicPartitions)`.  Otherwise LGTM assuming all tests pass.

Reply:
@rondagostino Thank you for the review!

The constructor of HashSet also states that it will NPE if the input is null.  Peeking at the  JDK source code, I see it does `Objects.requireNonNull` as part of its constructor.  So, there's fail-fast and fail-fast-enough.  I bucketed this as the latter, but I can add an explicit call if it is preferred in this method.  There is something to be said about 'how many times should the same thing be checked for a null value?'

https://docs.oracle.com/javase/8/docs/api/java/util/HashSet.html#HashSet-java.util.Collection-

Reply:
The overhead is minimal.  The benefit is clarity: if someone looks at the code it is immediately obvious that null is not allowed.